x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/softclassification/AucRoc.java

```diff
@@ -145,16 +146,23 @@ private String restLabelsAggName(ClassInfo classInfo) {
     public EvaluationMetricResult evaluate(ClassInfo classInfo, Aggregations aggs) {
         Filter classAgg = aggs.get(evaluatedLabelAggName(classInfo));
         Filter restAgg = aggs.get(restLabelsAggName(classInfo));
-        double[] tpPercentiles = percentilesArray(classAgg.getAggregations().get(PERCENTILES));
-        double[] fpPercentiles = percentilesArray(restAgg.getAggregations().get(PERCENTILES));
+        double[] tpPercentiles = percentilesArray(classAgg.getAggregations().get(PERCENTILES),
+            "[" + getMetricName() + "] requires at least one actual_field to have the value [" + classInfo.getName() + "]");
+        double[] fpPercentiles = percentilesArray(restAgg.getAggregations().get(PERCENTILES),
+            "[" + getMetricName() + "] requires at least one actual_field not to have the value [" + classInfo.getName() + "]");
```

Comment:
```suggestion
            "[" + getMetricName() + "] requires at least one actual_field NOT to have the value [" + classInfo.getName() + "]");
```

Reply:
I'm actually thinking it might be better to go with "... requires at least one actual_field to have a different value than [x]". I'll commit that.